include/zephyr/usb_c/usbc.h

```diff
@@ -220,6 +244,109 @@ typedef uint32_t (*policy_cb_get_rdo_t)(const struct device *dev);
  */
 typedef bool (*policy_cb_is_snk_at_default_t)(const struct device *dev);
 /** @}*/
+/** \addtogroup source_callbacks
+ *  @{
+ */
+/**
+ * @brief Callback type used to get the Rp value that should be placed on
+ *        the CC lines
+ *
+ * @param dev dev Runtime device structure
+ * @param rp
+ * @return 0 on success
+ */
+typedef int (*policy_cb_get_src_rp_t)(const struct device *dev,
+				      enum tc_rp_value *rp);
+
+/**
+ * @brief Callback type used to enable VBUS
+ *
+ * @param dev dev Runtime device structure
+ * @param en true to enable VBUS, else disable it
+ * @return 0 on success
+ */
+typedef int (*policy_cb_src_en_t)(const struct device *dev, bool en);
+
+/**
+ * @brief Callback type used to get the Source Capabilities
+ *	  from the Device Policy Manager
+ *
+ * @param dev dev Runtime device structure
+ * @param pdos pointer to source capability pdos
+ * @param num_pdos pointer to number of source capability pdos
+ * @return 0 on success
+ */
+typedef int (*policy_cb_get_src_caps_t)(const struct device *dev,
+					const uint32_t **pdos, uint32_t *num_pdos);
+
+/**
+ * @brief Callback type used to check if Sink request is valid
+ *
+ * @param dev dev Runtime device structure
+ * @param request_msg request message to check
+ * @return sink request reply
+ */
+typedef enum usbc_snk_req_reply_t (*policy_cb_check_sink_request_t)(
+				const struct device *dev,
+				const uint32_t request_msg);
```

Comment:
Looks like this file needs clang-format.

Reply:
done